functions/convert/main.py

```diff
@@ -0,0 +1,169 @@
+        self.errors = []
+
+    def close(self):
+        # delete temp files
+        if os.path.isdir(self.downloadDir):
+            shutil.rmtree(self.downloadDir, ignore_errors=True)
+        if os.path.isdir(self.filesDir):
+            shutil.rmtree(self.filesDir, ignore_errors=True)
+
+    def run(self):
+        global event, context
+
+        try:
+            # download the archive
+            fileToDownload = self.source_url
+            filename = self.source_url.rpartition('/')[2]
+            downloadedFile = os.path.join(self.downloadDir, filename)
+            try:
+                print('Downloading {0}...'.format(fileToDownload), end=' ')
+                if not os.path.isfile(downloadedFile):
+                    download_file(fileToDownload, downloadedFile)
+            finally:
+                print('finished.')
+
+            # unzip the archive
+            try:
+                print('Unzipping...'.format(downloadedFile), end=' ')
+                unzip(downloadedFile, self.filesDir)
+            finally:
+                print('finished.')
+
+            # create output directory
+            make_dir(self.outputDir)
+
+            # read the markdown files and output html files
+            try:
+                print('Processing the OBS markdown files')
+
+                files_to_process = sorted(glob(os.path.join(self.filesDir, '*.md')))
+
+                currentDir = os.path.dirname(os.path.realpath(__file__))
+                with codecs.open(os.path.join(currentDir, 'template.html'), 'r', 'utf-8-sig') as html_file:
+                    html_template = html_file.read()
+
+                completeHtml = ''
+                for filename in files_to_process:
+                   # read the markdown file
+                    with codecs.open(filename, 'r', 'utf-8-sig') as md_file:
+                        md = md_file.read()
+                        html = markdown.markdown(md)
+                        completeHtml += html
+                        html = TransformOBS.dirRe.sub(r'\1\n' + html + r'\n\2', html_template)
+                        htmlFileName = os.path.splitext(os.path.basename(filename))[0]+".html"
+                        write_file(os.path.join(self.outputDir, htmlFileName), html)
+
+                completeHtml = TransformOBS.dirRe.sub(r'\1\n' + completeHtml + r'\n\2', html_template)
+                write_file(os.path.join(self.outputDir, 'obs.html'), completeHtml)
+
+            except IOError as ioe:
+                print_error('{0}: {1}'.format(ioe.strerror, ioe.filename))
+                self.errors.append(ioe)
+
+            except Exception as e:
+                print_error(e.message)
+                self.errors.append(e)
+
+            finally:
+                print('finished.')
+
+        except Exception as e:
+            print_error(e.message)
+            self.errors.append(e)
+
+
+def handle(e, ctx):
+    global event, context, data
```

Comment:
Yeah, probably doesn't need to be global. I wasn't sure if I would need to refer to the payload in the transformOBS class, or other classes to come, as that will eventually have a link to a CSS file, and so forth, but maybe will check stick in in the options variable and not make it global. Also, the context from Lambda holds a unique ID that can be used, but not necessary. I'll change this to not use global variables. Thanks.